pkg/kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go

```diff
@@ -208,24 +211,35 @@ type muxRangeFeedEventReceiver interface {
 // startSingleRangeFeed looks up routing information for the
 // span, and begins execution of rangefeed.
 func (m *rangefeedMuxer) startSingleRangeFeed(
-	ctx context.Context, rs roachpb.RSpan, startAfter hlc.Timestamp, token rangecache.EvictionToken,
+	ctx context.Context,
+	rs roachpb.RSpan,
+	startAfter hlc.Timestamp,
+	token rangecache.EvictionToken,
+	fromManualSplit bool,
 ) error {
 	// Bound the partial rangefeed to the partial span.
 	span := rs.AsRawSpanWithNoLocals()
 
 	// Register active mux range feed.
 	stream := &activeMuxRangeFeed{
-		activeRangeFeed: newActiveRangeFeed(span, startAfter, m.registry, m.metrics),
+		// TODO(msbutler): it's sad that theres a bunch of repeat metadata.
+		// Deduplicate once old style rangefeed code is banished from the codebase.
+		activeRangeFeed: newActiveRangeFeed(span, startAfter, m.registry, m.metrics, fromManualSplit),
 		rSpan:           rs,
 		startAfter:      startAfter,
 		token:           token,
+		fromManualSplit: fromManualSplit,
 	}
 
 	if err := stream.start(ctx, m); err != nil {
 		stream.release()
 		return err
 	}
 
+	if m.cfg.withMetadata {
+		sendMetadata(m.eventCh, span, fromManualSplit)
+	}
+
```

Comment:
Does this race with stream.start()? Should we do this before starting the stream? If not, a comment on why not might be good. I'm assuming that we've decided we don't want to send this split event in the case where the stream was going to fail to start anyways?

Reply:
This is a good point. I'll add a comment explaining that i wanted to wait for the stream to successfully start.